autogen.sh

```diff
@@ -1,57 +1,46 @@
-#    may be used to endorse or promote products derived from this software
-#    without specific prior written permission.
+# The above copyright notice and this permission notice shall be included in
+# all copies of the Software and its documentation and acknowledgment shall be
+# given in the documentation and software packages that this Software was
+# used.
 #
-# THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
-# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
-# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
-# ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
-# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
-# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
-# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
-# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
-# SUCH DAMAGE.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-
-# Use clean option
-if [ "$1" = "clean" ] && [ ! -e Makefile ]
-then
-    echo "Vanishing the code"
-    rm -rf aclocal.m4 autom4te.cache/ compile configure depcomp install-sh \
-           Makefile.in missing src/config.h.in src/Makefile.in
-    exit 0
-fi
-
-# Do not use clean option
-if [ "$1" = "clean" ] && [ -e Makefile ]
-then
-    echo "I can not clean. Use '$ make distclean'."
-    exit 0
-fi
-
-# Do autoreconf
-autoreconf -i \
-   && { echo " "; \
-        echo "Done. You can use the 'clean' option to vanish the source code."; \
-        echo "Example of use: $ ./autogen.sh clean"; \
-        echo " "; \
-        echo "Now run ./configure, make, and make install."; \
-      } \
-|| { echo "We have a problem..."; exit 1; }
+case "$*" in
+    clean)
+        if [ -e Makefile ]; then
+            printf "Cannot clean. Run \`make distclean\` first.\n" >&2
+            exit 1
+        else
+            printf "Cleaning up.\n"
+            rm -rf aclocal.m4 autom4te.cache/ compile configure depcomp \
+                   install-sh Makefile.in missing src/config.h.in       \
+                   src/Makefile.in
+        fi
+        ;;
+    '')
+        autoreconf -i
+        printf '%s: done.\n' "${0##*/}"
+        printf "Run '\$ ./configure && make && make install' to install.\n"
+        printf "Or run '\$ ./autogen.sh clean' to clean up.\n"
```

Comment:
IIRC, this never really worked. If you ran `./autogen.sh` and then tried to run `./autogen.sh clean` then it will just tell you to run `make distclean` first.

Reply:
Actually my bad, it wouldn't fail if you run `./autogen.sh clean` right after. But it'd fail if you had ran `./configure`.